### DISCUSSION TITLE
Is it possible to extract a source from the stylesheet for manipulation?

### DISCUSSION THREAD

**Discussion by Tasemu** [rnmapbox/maps#2973] [Q&A]
Currently i have a geojson source i'm fetching in code and injecting into the map. I was hoping instead i could add this source into the stylesheet using mapbox studio, and then using rnmapbox, extract the source to work with in my code. The main reason for this is that every user downloading my geojson file from firebase can get very data-hungry.

Thanks in advance for any and all advice! 👏

**Comment by Tasemu** [comment]
Bump? :)

**Reply by mfazekas** [reply]
Hi, yes, you can refer to an existing source in style with the `id={...}` and using the `existing` attribute as well.

**Reply by Tasemu** [reply]
Sorry mate, i've been looking through the docs but can't see what you're referring to, i get the gist of what you mean but can't think of how i'd implement it. If possible is there any chance you could give me an example of getting a source and assigning it to a variable? I'm probably missing a simple point here but it'd be really useful. Cheers if possible!

**Reply by mfazekas** [reply]
See ShowAndHideLayer for an example, that's what we have, it might now cover your use case.

**Comment by Tasemu** [comment]
> See ShowAndHideLayer for an example, that's what we have, it might now cover your use case.

I had a look at the Mapbox api and have tried the following as a use case, however there doesn't appear to be a method to get the MapboxMap. I had hoped once I got this instance i'd be able to retrieve the source with the id. Is there another way i'm supposed to access the style that i'm not seeing?

**Reply by Tasemu** [reply]
```
useEffect(() => {
    if (mapRef.current) {
      console.log(mapRef.current.mapboxMap.style);
    }
  }, [mapRef.current]);
```
